lib/openstack_query/query_blocks/query_executer.py

```diff
@@ -96,37 +127,89 @@ def run_query(
             cloud_account = cloud_account.name.lower()
 
         start = time.time()
-        self.raw_results = self.runner.run(
+        results = self.runner.run(
             cloud_account=cloud_account,
             client_side_filters=self.client_side_filters,
             server_side_filters=self.server_side_filters,
             from_subset=from_subset,
             **kwargs,
         )
+
+        forwarded_results = copy.deepcopy(self._forwarded_values)
+        self.results = self.apply_forwarded_results(
+            results, forwarded_results, self.link_prop_func
+        )
```

Comment:
I don't see why we need to do the copies if we have a list of objects?
I was thinking you simply have something like:

```
if not results:
   # First query
   self.results = [() for _ in len(query_results)

for dest_tuple, openstack_obj in zip(self.results, query_results)
   dest_tuple.append(openstack_obj)
```

since `append_forward_results` is a bit of a beast currently:

Reply:
To handle the removal case which would look something like:

```
new_results = []

for dest_tuple, openstack_obj in zip(self.results, query_results)
   was_removed = False
   for obj in dest_tuple:
        # check if any of the fields match, if none do we can assume this tuple was removed
        if not any(getattr(field_name, obj) == getattr(field_name, openstack_obj) for field_name in openstack_obj):
             continue

   dest_tuple.append(openstack_obj)
   new_results.append(dest_tuple)
self.results = new_results
```